darwin.c

```diff
@@ -7,82 +7,8 @@
 #include <mach-o/dyld.h>
 #include "common.h"
 
-int elevate(char *path, char *prompt, char *iconPath)
-{
-  AuthorizationEnvironment authEnv;
-  AuthorizationItem kAuthEnv[2];
-  authEnv.items = kAuthEnv;
-  authEnv.count = 0;
-
-  if (prompt != NULL) {
-    kAuthEnv[authEnv.count].name = kAuthorizationEnvironmentPrompt;
-    kAuthEnv[authEnv.count].valueLength = strlen(prompt);
-    kAuthEnv[authEnv.count].value = prompt;
-    kAuthEnv[authEnv.count].flags = 0;
-    authEnv.count++;
-  }
-  if (iconPath != NULL) {
-    kAuthEnv[authEnv.count].name = kAuthorizationEnvironmentIcon;
-    kAuthEnv[authEnv.count].valueLength = strlen(iconPath);
-    kAuthEnv[authEnv.count].value = iconPath;
-    kAuthEnv[authEnv.count].flags = 0;
-    authEnv.count++;
-  }
-
-  AuthorizationItem authItems[1];
-  authItems[0].name = kAuthorizationRightExecute;
-  authItems[0].valueLength = 0;
-  authItems[0].value = NULL;
-  authItems[0].flags = 0;
-
-  AuthorizationRights authRights;
-  authRights.count = sizeof(authItems) / sizeof(authItems[0]);
-  authRights.items = authItems;
-
-  AuthorizationFlags authFlags;
-  authFlags = kAuthorizationFlagDefaults | kAuthorizationFlagInteractionAllowed | kAuthorizationFlagExtendRights;
-
-  AuthorizationRef authRef;
-  OSStatus status = AuthorizationCreate(&authRights, &authEnv, authFlags, &authRef);
-  if(status != errAuthorizationSuccess) {
-    NSLog(@"Error create authorization");
-    return NO_PERMISSION;
-  }
-
-  FILE *pipe = NULL;
-  char* argv[] = { "setuid", NULL };
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-  status = AuthorizationExecuteWithPrivileges(authRef, path, kAuthorizationFlagDefaults, argv, &pipe);
-#pragma GCC diagnostic warning "-Wdeprecated-declarations"
-  if(status != errAuthorizationSuccess) {
-    NSLog(@"Error run %s with privileges: %d", path, status);
```

Comment:
If AuthorizationExecuteWithPrivileges failed because the tool experienced a failure, this loses the output from the tool.  cocoasudo does not exhibit the same bug.